Bootstrapping bundle init

PrimalPimmy opened the pull request.

PrimalPimmy committed:
Bootstrapping bundle init

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

PrimalPimmy committed:
minor fix

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

shivaccuknox commented:
Please attach  the call flow diagram to the PR so that reviewers will have some context

PrimalPimmy committed:
Cluster reconciler

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

PrimalPimmy committed:
Cluster reconciler

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

PrimalPimmy committed:
sending bundle to remote cluster

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

PrimalPimmy committed:
Test

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

PrimalPimmy committed:
Spire auth inside reconciler

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

shivaccuknox commented on controllers/pkg/reconcilers/spire-bootstrap/reconciler.go:
```diff
@@ -0,0 +1,287 @@
+
+	// // Example: Update the status if necessary
+
+	socketPath := "unix:///spiffe-workload-api/agent.sock"
```
Create a config map for socket path, Vault IP address in nephio-system ns.

shivaccuknox commented on controllers/pkg/reconcilers/spire-bootstrap/reconciler.go:
```diff
@@ -0,0 +1,287 @@
+	audience := "TESTING"
+	spiffeID := spiffeid.RequireFromString("spiffe://example.org/nephio")
+
+	jwtSVID, err := jwtSource.FetchJWTSVID(ctx, jwtsvid.Params{
```
Add a function to get a jwt token (for current workload)

shivaccuknox commented on controllers/pkg/reconcilers/spire-bootstrap/reconciler.go:
```diff
@@ -0,0 +1,287 @@
+	// Set the client token
+	client.SetToken(clientToken)
+
+	// Retrieve the secret
```
Create a function to retrieve secrets. Also, create a function to write library

PrimalPimmy committed:
Vault authentication and fetching kubeconfig

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

PrimalPimmy commented:
Ref: https://github.com/5GSEC/SIG-Security/issues/9

PrimalPimmy committed:
Some testing

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

PrimalPimmy committed:
Mid testing

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Mid testing

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Mid testing

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Mid testing-2

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Mid testing-2

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Mid testing-2

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Mid testing-3

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Mid testing-3

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Mid testing-3

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

PrimalPimmy committed:
Refactoring

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Kubeconfig test

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Removed found

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

log testing

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

log testing

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

log testing

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

PrimalPimmy commented:
TODOs:
Create configmap/env variables for things like Vault IP, Spire unix domain socket, etc

PrimalPimmy committed:
Getting svid

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Getting svid-2

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Getting svid-2

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Getting svid-3

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Getting svid-4

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Getting svid-5

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Getting svid-6

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

PrimalPimmy committed:
Vault addr and patch configmap

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Vault addr and patch configmap

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

PrimalPimmy committed:
Restricted Kubeconfig

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

configmap perms

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

server addr

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Struct to yaml

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Struct to yaml

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Reverting changes

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Reverting changes

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Reverting changes

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Reverting changes

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Reverting changes

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Reverting changes

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Reverting changes

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

Reverting changes

Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

shivaccuknox reviewed (approved):
There are some changes need to this commit. Will raise separate issues for tracking them
